src/adapter/src/catalog/storage.rs

```diff
@@ -1053,6 +1054,50 @@ impl<'a> Transaction<'a> {
         Ok(())
     }
 
+    pub fn rename_cluster(
+        &mut self,
+        cluster_id: ClusterId,
+        cluster_name: &str,
+        cluster_to_name: &str,
+    ) -> Result<(), Error> {
+        let key = ClusterKey { id: cluster_id };
+
+        match self.clusters.update(|k, v| {
+            if *k == key {
+                let mut value = v.clone();
+                value.name = cluster_to_name.to_string();
+                Some(value)
+            } else {
+                None
+            }
+        })? {
+            1 => Ok(()),
+            _ => Err(SqlCatalogError::UnknownCluster(cluster_name.to_string()).into()),
```

Comment:
Untested error case

Reply:
This case can only happen when the catalog state is out-of-sync, which is not a case I can test. I'm not sure that there is any benefit in returning this error over panicking? Maybe @jkosh44 has an opinion here.